fms2_io/netcdf_io.F90

```diff
@@ -567,59 +567,83 @@ function netcdf_file_open(fileobj, path, mode, nc_format, pelist, is_restart, do
                                               !! is a restart file.  Defaults
                                               !! to false.
   logical, intent(in), optional :: dont_add_res_to_filename !< Flag indicating not to add
-                                              !! ".res" to the filename
+                                                            !! ".res" to the filename
+  integer, intent(in), optional :: tile_comm      !< MPI communicator used for parallel I/O. Passing this
+                                                  !! argument enables parallel I/O.
+  logical, intent(in), optional :: use_collective !< Flag indicating whether reads and writes should be performed
+                                                  !! collectively rather than independently.
   logical :: success
 
   integer :: nc_format_param
   integer :: err
   integer :: netcdf4 !< Query the file for the _IsNetcdf4 global attribute in the event
                      !! that the open for collective reads fails
-  character(len=FMS_PATH_LEN) :: buf !< File path with .res in the filename if it is a restart
-  character(len=FMS_PATH_LEN) :: buf2 !< File path with the filename appendix if there is one
+  character(len=FMS_PATH_LEN) :: full_path !< File path with .res in the filename if it is a restart
   logical :: is_res
   logical :: dont_add_res !< flag indicated to not add ".res" to the filename
 
+  success = .true.
+
+  if (fms2_nc_format_param.eq.-1) then
+    call error("netcdf_file_open :: fms2_io has not been initialized")
+  endif
+
   if (allocated(fileobj%is_open)) then
-    if (fileobj%is_open) then
-      success = .true.
-      return
-    endif
+    if (fileobj%is_open) return
+  endif
+
+  fileobj%use_netcdf_mpi = .false.
+
+  if (fileobj%tile_comm.ne.MPP_COMM_NULL) then
+    call mpp_error(NOTE, "netcdf_file_open :: Setting fileobj%tile_comm is deprecated. &
+                          Please use open_file(..., tile_comm=...) instead.")
+    fileobj%use_netcdf_mpi = .true.
+  elseif (present(tile_comm)) then
+    fileobj%use_netcdf_mpi = .true.
+    fileobj%tile_comm = tile_comm
+  endif
+
+  if (fileobj%use_collective) then
+    fileobj%use_netcdf_mpi = .true.
+    call mpp_error(NOTE, "Setting fileobj%use_collective to enable collective reads is deprecated. &
+                          Please use open_file(..., use_collective=.true.) instead.")
+  else
+    fileobj%use_collective = .false.
+    if (present(use_collective)) fileobj%use_collective = use_collective
   endif
+
```

Comment:
I think we don't need to the else statement here.

Before this update to use collective io, they will do:
```
fileobj%use_collective = .True.
fileobj%tile_comm = ...
open_file(fileobj, ...)
```

Now they will just need to do
```
open_file(fileobj, ..., use_netcdf_mpi=.True.
```
use_netcdf_mpi and use_collective are equivalent.

So to not break, their current code we should just do:
```
  if (fileobj%use_collective) then
    fileobj%use_netcdf_mpi = .true.
    call mpp_error(NOTE, "Setting fileobj%use_collective to enable collective reads is deprecated. &
                          Please use open_file(..., use_collective=.true.) instead.")
endif
```

and remove any use of `use_collective` in the code.